### PR TITLE
112 a round may have a different deadline for staff than external users

### PR DESCRIPTION
**Pull request by mutambaraf**
## Description

A round may have a different deadline for staff than external users.

For example, the ISIS direct deadline is 1 week later for staff as they provide support and advice to users during the user submission period.

## Motivation and Context

These changes will allow internal users to submit proposals after call close when it is set in the call through a call internal call end date and including internal editable in the proposal workflow.

## How Has This Been Tested

E2E test have been added

## Fixes
https://github.com/UserOfficeProject/user-office-project-issue-tracker/issues/112

## Changes

- Add additional field for internal call end date and its related flag in database.
- Add proposal workflow status for editable submitted internal proposals.
- Add call internal event to be emitted when internal call ends.
- Update existing call end asynchronous job for call internal ended check and proposal update.
- Add a new filter to calls for active internal calls.
- Add new internal user identifier flag to user JWT payload back and front end,
- Add internal call event for internal call close.

## Depends on

Validation of internal call end date is included in this PR
https://github.com/UserOfficeProject/user-office-lib/pull/155

## Tests included/Docs Updated?

- [x] I have added tests to cover my changes.
- [ ] All relevant doc has been updated

### CODE REVIEW

**Review by EdwardHaynes — APPROVED** [review]

**Comment by mutambaraf** [comment]
Have added change requested from sprint meeting
image

File: apps/user-office-backend/src/events/event.enum.ts [hunk 1]
```diff
@@ -157,6 +158,10 @@ export const EventLabel = new Map<Event, string>([
     Event.CALL_ENDED,
     'Event occurs on a specific call end date set on the call',
   ],
+  [
+    Event.CALL_ENDED_INTERNAL,
+    'Event occurs on a specific call  internal end date set on the call',
```
Comment (jekabs-karklins) [inline apps/user-office-backend/src/events/event.enum.ts:163]:
```suggestion
    'Event occurs on a specific call internal end date set on the call',
```

**Review by jekabs-karklins — COMMENTED** [review]
I had a few comments and please check if this PR can be merged with changes I already did
https://github.com/UserOfficeProject/user-office-core/pull/75

File: apps/user-office-backend/src/auth/UserAuthorization.ts [hunk 2]
```diff
@@ -22,6 +22,20 @@ export abstract class UserAuthorization {
     return agent?.currentRole?.shortCode === Roles.USER_OFFICER;
   }
 
+  async isInternalUser(agent: UserWithRole | null): Promise<boolean> {
+    return agent
+      ? agent.currentRole?.shortCode === Roles.INSTRUMENT_SCIENTIST ||
+          agent.currentRole?.shortCode === Roles.USER_OFFICER ||
+          this.userDataSource.getRolesForUser(agent.id).then((roles) => {
```
[the remark this answers is not in the document]
Comment (mutambaraf) [reply apps/user-office-backend/src/auth/UserAuthorization.ts:29]: Will resolve this when l merge from develop

**Review by jekabs-karklins — APPROVED** [review]

**Pull request merged**